File: electrician-management/src/main/java/com/electrician/electrician_management/service/ElectricianService.java
package com.electrician.electrician_management.service;

import com.electrician.electrician_management.Entity.Electrician;

import java.util.List;

public interface ElectricianService {

    //Admin use
    Electrician addElectrician(Electrician electrician);

    //electrician login
    Electrician loginByPhone(String phone);

    //electrician use
    Electrician subscribePlan(Long electricianId, Long planId);

    List<Electrician> getAllElectricians();
    List<Electrician> getActiveElectricians();
    List<Electrician> getExpiredElectricians();

    Electrician getById(Long id);



}

File: electrician-frontend/src/app/core/services/electrician.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';
import { Observable } from 'rxjs';

export interface SubscriptionPlan {
  id: number;
  planName: string;
  durationDays: number;
  price: number;
}

export interface Electrician {
  id: number;
  name: string;
  phone: string;
  status: string;
  membershipEnd: string;
  subscriptionPlan?: SubscriptionPlan;
}

@Injectable({ providedIn: 'root' })
export class ElectricianService {

  private baseUrl = environment.apiUrl + '/admin/electrician';

  constructor(private http: HttpClient) {}

  getAllElectricians(): Observable<Electrician[]> {
    return this.http.get<Electrician[]>(`${this.baseUrl}/all`);
  }

  addElectrician(data: any) {
    return this.http.post(`${this.baseUrl}/add`, data);
  }

  subscribeElectrician(electricianId: number, planId: number) {
    return this.http.post(
      `${this.baseUrl}/subscribe/${electricianId}/${planId}`,
      {}
    );
  }
}